src/js/d3ViewComponent.js

```diff
@@ -56,26 +56,46 @@ https://github.com/gpii/universal/LICENSE.txt
 
     // Synthesize "styles" and "selectors" blocks to combine elements with the same key
     gpii.d3ViewComponent.synthesizeClasses = function (styles, selectors) {
-        var result = {};
+
+        var consolidatedClasses = {};
+
+        // 1. Combine any matching styles and selectors by key into a single string of class names
 
         fluid.each(styles, function (styleValue, key) {
-            var correspondingSelector = fluid.get(selectors, key),
-                selectorName = gpii.d3ViewComponent.extractSelectorName(correspondingSelector),
-                concatenated = selectorName ? selectorName.concat(" " + styleValue) : styleValue;
-            fluid.set(result, key, concatenated);
+            fluid.set(consolidatedClasses, key, styleValue);
         });
 
         fluid.each(selectors, function (selector, key) {
-            var correspondingStyle = fluid.get(styles, key);
-            if (!correspondingStyle) {
-                var selectorName = gpii.d3ViewComponent.extractSelectorName(selector);
-                if (selectorName) {
-                    fluid.set(result, key, selectorName);
-                }
-            }
+            var correspondingStyle = fluid.get(consolidatedClasses, key);
+            if(correspondingStyle) {
+                var combinedValues = consolidatedClasses[key] + " " + gpii.d3ViewComponent.extractSelectorName(selector);
+                fluid.set(consolidatedClasses, key, combinedValues);
+            } else {fluid.set(consolidatedClasses, key, gpii.d3ViewComponent.extractSelectorName(selector));}
+        });
+
+        // 2. For each key in the consolidatedClasses object, split its value into an array of string values separated by spaces,
+        // filter for unique strings, and make a new value based on the array of uniques
+
+        fluid.each(consolidatedClasses, function(classes, key) {
+            var splitClasses = classes.split(" ");
+            // Basic alphabetic sort to improve readability of applied classes and make testing easier
+            splitClasses.sort(function (a,b) {
```

Comment:
Infusion's convention for applying selectors to DOM elements are, control selector names, the ones starting with "gpiic-" in the chart authoring context, comes before styling selector names that start with "gpii-". The convention was decided 'cuz usually a DOM element only has one control selector but several styling selectors. Would be nice it the consolidated classes can sustain this convention.